drivers/gke/gke_driver.go

```diff
@@ -324,14 +466,64 @@ func getStateFromOpts(driverOptions *types.DriverOptions) (state, error) {
 		d.Locations = append(d.Locations, location)
 	}
 
-	d.NodeCount = options.GetValueFromDriverOptions(driverOptions, types.IntType, "node-count", "nodeCount").(int64)
 	labelValues := options.GetValueFromDriverOptions(driverOptions, types.StringSliceType, "labels").(*types.StringSlice)
 	for _, part := range labelValues.Value {
 		kv := strings.Split(part, "=")
 		if len(kv) == 2 {
-			d.NodeConfig.Labels[kv[0]] = kv[1]
+			d.NodePool.Config.Labels[kv[0]] = kv[1]
+		}
+	}
+	resourceLabelValues := options.GetValueFromDriverOptions(driverOptions, types.StringSliceType, "resource-labels", "resourceLabels").(*types.StringSlice)
+	for _, part := range resourceLabelValues.Value {
+		kv := strings.Split(part, "=")
+		if len(kv) == 2 {
+			d.ResourceLabels[kv[0]] = kv[1]
+		}
+	}
+
+	oauthScopesValues := options.GetValueFromDriverOptions(driverOptions, types.StringSliceType, "oauth-scopes", "oauthScopes").(*types.StringSlice)
+	for _, oauthScope := range oauthScopesValues.Value {
+		d.NodePool.Config.OauthScopes = append(d.NodePool.Config.OauthScopes, oauthScope)
+	}
+
+	// Configuration of NodePoolAutoscaling
+	d.NodePool.Autoscaling.Enabled = options.GetValueFromDriverOptions(driverOptions, types.BoolType, "enable-nodepool-autoscaling", "enableNodepoolAutoscaling").(bool)
+	d.NodePool.Autoscaling.MinNodeCount = options.GetValueFromDriverOptions(driverOptions, types.IntType, "min-node-count", "minNodeCount").(int64)
+	d.NodePool.Autoscaling.MaxNodeCount = options.GetValueFromDriverOptions(driverOptions, types.IntType, "max-node-count", "maxNodeCount").(int64)
+	d.NodePool.Name = "default-0"
```

Comment:
Is this a fixed name? Does it conflict if multiple clusters/node pools are created?

Reply:
yes, it is a fixed name, I think it will not conflict because this is the first node pool and the name is required, as well GKE will not generate a same node pool name when we are creating another node pool.